### PR TITLE
feat: update docs

### LINKED ISSUE #60: Setup fails if I don't install cocoapods
#### Issue description
While using this template, it asks if I want to install cocoapods if I don't already have it installed. However, if it fails, then the setup fails with an unhandled promise rejection.

#### Steps to reproduce the issue

1.  Make sure you don't have cocoapods installed 
2. Run `react-native init MyApp --template=https://github.com/echobind/react-native-template-echobind`
3. Select "yes" to install cocoapods
4. Type in your password incorrectly
5. Observe promise rejection error

#### What's the expected result?
I should be able to retype in my password, or restart the whole process

#### What's the actual result?
Promise rejection warning.

#### Additional details / screenshot
image

### LINKED ISSUE #69: update the docs
- new format
- add eb logo
- add a section about tech used and why
- update the setup instructions

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR revamps the README and adds an MIT `LICENSE.md`

## Changes
- update format of README
- update setup instructions
- add eb logo
- add MIT license

## Screenshots
(prefer animated gif)

## Checklist
- [ ] Automated tests
- [ ] Creating a new project still works
- [X] Added docs

Fixes #69 
Fixes #60 (tells user to install cocoapods during setup)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,11 +99,27 @@ Our Recommended template for React Native projects.
...
+
+## CI / Deployment:
+
+In order to set up CI/Deployment, we've created custom hygen templates to save you time. These live in our [`hygen-templates-echobind` repository](https://github.com/echobind/hygen-echobind-templates). You'll find installation instructions, which will allow you to set up the following:
+
+- CircleCI preconfigured for React Native. Fast build times!
+- Fastlane lanes for alpha, beta, and production releases. These should be run on CI only!
+- AppCenter integration for easy alpha distribution
+
+For more info, please follow the [instructions](https://github.com/echobind/hygen-echobind-templates) in that repo.
```
Comment (cball) [inline README.md]: looks like we also have some CI docs in `docs/circle-ci`. Should we move some of this there? What's the relation to the local hygen templates (in `_templates`) vs the `hygen-echobind-templates` repo?

File: README.md [hunk 2]
```diff
@@ -1,17 +1,62 @@
...
+    <img src="https://camo.githubusercontent.com/d22763c73585cf5d4cf87534659689c2a6b3f214/68747470733a2f2f7265732d332e636c6f7564696e6172792e636f6d2f6372756e6368626173652d70726f64756374696f6e2f696d6167652f75706c6f61642f635f6c7061642c685f3235362c775f3235362c665f6175746f2c715f6175746f3a65636f2f76313439393437333135312f68326b3233696f6f3479687230676a746f636d792e6a7067" alt="Logo" width="80" height="80">
+  </a>
+
+  <h3 align="center">react-native-template-echobind</h3>
+
+  <p align="center">
+    Our Recommended template for React Native projects.
+    <br />
+    <br />
+    <a href="https://github.com/othneildrew/Best-README-Template">View Demo</a>
```
Comment (cball) [inline README.md]: should we remove this until we have a video or something else to link to?

Comment (jsjoeio) [reply README.md]: Oops...yes, I'll remove

File: README.md (hunk 1 above)
Comment (jsjoeio) [reply README.md]: Good points! I address some of this in PR #56 - I'll go there and make an update related to `docs/circle-ci` as well.

**Pull request merged**